capstone/src/main/webapp/youtubeScript.js

```diff
@@ -86,28 +87,44 @@ async function inputCommentsToPerspective(commentsList) {
   if (!requestedAttributes) {
       return;
   }
-  const attributeTotals = new Map();
+  const attributeScoresPromises = [];
   for (const comments in commentsList) {
-    const attributeScores = [];
     for (const item in commentsList[comments].items) {
       const perspectiveScore = await callPerspective(commentsList[comments].items[item].snippet.topLevelComment.snippet.textOriginal, langElement.value, requestedAttributes);
-      attributeScores.push(perspectiveScore);
+      attributeScoresPromises.push(perspectiveScore);
     }
-    for (var i = 0; i < requestedAttributes.length; i++) {
-      for (var j = 0; j < attributeScores.length; j++) {
-        if (attributeTotals.has(requestedAttributes[i])) {
-          attributeTotals.set(requestedAttributes[i], attributeTotals.get(requestedAttributes[i]) + attributeScores[j].attributeScores[requestedAttributes[i]].summaryScore.value);
-        } else {
-          attributeTotals.set(requestedAttributes[i], attributeScores[j].attributeScores[requestedAttributes[i]].summaryScore.value);
-        }
+  }
+  await Promise.all(attributeScoresPromises).then(resolvedAttributeScores => {
+    const attributeTotals = getAttributeTotals(resolvedAttributeScores);
+    const attributeAverages = getAttributeAverages(attributeTotals, commentsList);
+    loadChartsApi(attributeAverages);
+    perspectiveToxicityScale(attributeAverages);
+  });
+}
+
+/** Returns a map of attribute score sums from an array of JSON's */
+function getAttributeTotals(attributeScores) {
+  const requestedAttributes = getRequestedAttributes();
+  const attributeTotals = new Map();    
+  for (let i = 0; i < requestedAttributes.length; i++) {
+    for (let j = 0; j < attributeScores.length; j++) {
+      if (attributeTotals.has(requestedAttributes[i])) {
+        attributeTotals.set(requestedAttributes[i], attributeTotals.get(requestedAttributes[i]) + attributeScores[j].attributeScores[requestedAttributes[i]].summaryScore.value);
+      } else {
+        attributeTotals.set(requestedAttributes[i], attributeScores[j].attributeScores[requestedAttributes[i]].summaryScore.value);
```

Comment:
Same note as [this comment](https://github.com/googleinterns/step40-2020/pull/29#discussion_r459720932), we might want to create local variables to increase readability here

Reply:
addressed in "formatting improvements"